pyswarms/backend/topology/pyramid.py

```diff
@@ -0,0 +1,133 @@
+# -*- coding: utf-8 -*-
+
+"""
+A Pyramid Network Topology
+
+This class implements a star topology where all particles are connected in a
+pyramid like fashion.
+"""
+
+# Import from stdlib
+import logging
+
+# Import modules
+import numpy as np
+from scipy.spatial import Delaunay
+
+# Import from package
+from .. import operators as ops
+from .base import Topology
+
+# Create a logger
+logger = logging.getLogger(__name__)
+
+class Pyramid(Topology):
+    def __init__(self):
+        super(Pyramid, self).__init__()
+    
+    def compute_gbest(self, swarm):
+        """Updates the global best using a pyramid neighborhood approach
+
+        This uses the Delaunay method from :code:`scipy` to triangulate space
+        with simplices
+
+        Parameters
+        ----------
+        swarm : pyswarms.backend.swarms.Swarm
+            a Swarm instance
+
+        Returns
+        -------
+        numpy.ndarray
+            Best position of shape :code:`(n_dimensions, )`
+        float
+            Best cost
+        """
+        try:
+            # If there are less than 5 particles they are all connected
+            if swarm.n_particles < 5:
+                best_pos = swarm.pbest_pos[np.argmin(swarm.pbest_cost)]
+                best_cost = np.min(swarm.pbest_cost)
+            else:
+                pyramid = Delaunay(swarm.position)
+                indices, indptr = pyramid.vertex_neighbor_vertices
+                idx = np.array()
+                # Insert all the neighbors for each particle in the idx array
+                for i in range(swarm.n_particles):
+                    idx.append(indptr[indices[i]:indices[i+1]])
+                idx_min = swarm.pbest_cost[idx].argmin(axis=1)
+                best_neighbor = idx[np.arange(len(idx)), idx_min]
+                
+                # Obtain best cost and position
+                best_cost = np.min(swarm.pbest_cost[best_neighbor])
+                best_pos = swarm.pbest_pos[
+                    np.argmin(swarm.pbest_cost[best_neighbor])
+                ]
+        except AttributeError:
+            msg = "Please pass a Swarm class. You passed {}".format(
+                type(swarm)
+            )
+            logger.error(msg)
+            raise
+        else:
+            return (best_pos, best_cost)
+            
+        def compute_velocity(self, swarm, clamp=None):
```

Comment:
Hey @whzup , can you check the indentation here? I think you should move this 4 spaces backward :+1: 
Right now, it is aligned *inside* `def compute_gbest()`, the three methods should be on the same level.
On my end, I wonder why it didn't raise any `NotImplementedError`...